Description: Further fixes to the build on the GNU Hurd
 - include <sys/file.h> for LOCK_EX
 - define MAXHOSTNAMELEN if absent
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2010-10-16

--- a/spool.c
+++ b/spool.c
@@ -32,6 +32,7 @@
  * SUCH DAMAGE.
  */
 
+#include <sys/file.h>
 #include <sys/stat.h>
 
 #include <ctype.h>
--- a/util.c
+++ b/util.c
@@ -33,6 +33,7 @@
  */
 
 #include <sys/param.h>
+#include <sys/file.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <netdb.h>
@@ -43,6 +44,14 @@
 
 #include "dma.h"
 
+/**
+ * A quick'n'dirty hack to get dma to build on the GNU Hurd.
+ * A real fix would dynamically allocate the hostname array.
+ */
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN	1024
+#endif
+
 const char *
 hostname(void)
 {
